nx-X11/programs/Xserver/dix/dixutils.c

```diff
@@ -241,15 +224,10 @@ SecurityLookupDrawable(XID rid, ClientPtr client, Mask access_mode)
 
     if(rid == INVALID)
 	return (void *) NULL;
-    if (client->trustLevel != XSecurityClientTrusted)
-	return (DrawablePtr)SecurityLookupIDByClass(client, rid, RC_DRAWABLE,
-						    access_mode);
-    if (client->lastDrawableID == rid)
-	return ((void *) client->lastDrawable);
     pDraw = (DrawablePtr)SecurityLookupIDByClass(client, rid, RC_DRAWABLE,
 						 access_mode);
-    if (pDraw && (pDraw->type != UNDRAWABLE_WINDOW))
-        return (void *)pDraw;
+    if ((client->trustLevel != XSecurityClientTrusted) || (pDraw && (pDraw->type != UNDRAWABLE_WINDOW)))
+	return (void *)pDraw;
```

Comment:
Yeah, this old X.Org version is riddled with mixed indentation and styles. The upstream folks ran a script to fix up whitespace in a consistent manner some time ago, so we'll need to do this as well. My point is aiming for the smallest diff, though (which may be a moot one since `git diff` supports the `-w` flag...)

As you like, really.